### PR TITLE
[DOCS][ESQL] Document locale rest parameter

### LINKED ISSUE #104671: ESQL: document locale rest parameter
### Description

ESQL Rest endpoints supports `locale` parameter and returns results (especially dates) according to it - this needs to be documented:
```
POST _esql
{
  "locale": "it_IT",
  "query": """
          ROW birth_date_string = "2023-01-15T00:00:00.000Z" 
          | EVAL birth_date = date_parse(birth_date_string)
          | EVAL month_of_birth = date_format(birth_date, "MMMM")
   """
}
```
/cc @luigidellaquila

### PR DESCRIPTION
**Pull request by leemthompo**
Closes https://github.com/elastic/elasticsearch/issues/104671 

- Wording is pretty drafty :)
- Had to play around to get this working and looks like we accept [ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes)

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]
Fine by me.

File: docs/reference/esql/esql-rest.asciidoc [hunk 1]
```diff
@@ -204,6 +204,30 @@ Which returns:
 }
 ----
 
+[discrete]
+[[esql-locale-param]]
+==== Returning localized results
+
+Use the `locale` parameter in the request body to return results (especially dates) formatted per the conventions of the locale.
+If `locale` is not specified, defaults to `en` (English).
```
Comment (astefan) [inline docs/reference/esql/esql-rest.asciidoc:212]: The default locale that ESQL uses is `en_US`.

Comment (costin) [reply docs/reference/esql/esql-rest.asciidoc:212]: ISO-639 is just part of the solution - the JDK uses other ISOs as indicated [here](https://oracle.com/java/technologies/javase/jdk17-suported-locales.html). This also shows up in the default `en_US` (lang + country).

Comment (leemthompo) [reply docs/reference/esql/esql-rest.asciidoc:212]: In terms of usage, I actually couldn't get **locale** to work with codes of the form `xy_XY`, only `xy`.

Comment (leemthompo) [reply docs/reference/esql/esql-rest.asciidoc:212]: @astefan @costin 

The `locale` parameter works with parameters of format (case-insensitive)`xy` and `xy-XY`, but not `xy_XY`

Comment (astefan) [reply docs/reference/esql/esql-rest.asciidoc]: 👍 you are correct, the parameter is a `LanguageTag` instance. And from https://www.oracle.com/java/technologies/javase/jdk17-suported-locales.html the Language Tag column shows `en-US`.

Comment (leemthompo) [reply docs/reference/esql/esql-rest.asciidoc]: I guess we can backport this to 8.12?

**Review by costin — APPROVED** [review]
LGTM.
👍  for backporting.

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**